### PR TITLE
DEV: Add linting and CI config

### PR DESCRIPTION
**Pull request by davidtaylorhq**

### CODE REVIEW

File: .eslintrc
Comment (CvX) [inline .eslintrc:1]: Shall we use the new linting setup? 😃 (I'll update the skeleton shortly)

Comment (davidtaylorhq) [reply .eslintrc:1]: Sure, I guess discourse-gifs is a good one to copy for now?

Comment (CvX) [reply .eslintrc:1]: yup, and discourse-assign for plugins (https://github.com/discourse/discourse-assign/pull/517)

**Comment by davidtaylorhq** [comment]
(backend test failure is expected, and will be resolved by #2)

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+source "https://rubygems.org"
+
+group :development do
+  gem "rubocop-discourse"
+  gem "syntax_tree"
+  gem "syntax_tree-disable_ternary"
```
Comment (CvX) [inline Gemfile:8]:
```suggestion
```

**Review by CvX — APPROVED** [review]

**Pull request merged**